### PR TITLE
Add all global parameters description

### LINKED ISSUE #1005: Missing postprocess parameter in global default.yaml file
## **Desired feature/enhancement**

Add the description of the `postprocess` parameter in the global default.yaml file

## **Motivation**

`postprocess` is an global parameter that can be defined in a config file (default is true), but there is no description of what it does in the default.yaml file

## **Description**

Add the missing entry in `src/haddock/modules/default.yaml`

### PR DESCRIPTION
**Pull request by VGPReys**
You are about to submit a new Pull Request. Before continuing make sure you read the [contributing guidelines](CONTRIBUTING.md) and that you comply with the following criteria:

- [X] You have sticked to Python. Please talk to us before adding other programming languages to HADDOCK3
- [ ] Your PR is about CNS
- [X] Your code is well documented: proper docstrings and explanatory comments for those tricky parts
- [X] You structured the code into small functions as much as possible. You can use classes if there is a (state) purpose
- [X] Your code follows our coding style
- [ ] You wrote tests for the new code
- [X] `tox` tests pass. *Run `tox` command inside the repository folder*
- [ ] `-test.cfg` examples execute without errors. *Inside `examples/` run `python run_tests.py -b`*
- [X] PR does not add any dependencies, unless permission granted by the HADDOCK team
- [X] PR does not break licensing
- [ ] Your PR is about writing documentation for already existing code :fire:
- [ ] Your PR is about writing tests for already existing code :godmode:

---

Closes #1005 .

- Had to add the `optional.yaml` and `mandatory.yaml` parsing when no optional parameters in `haddock3-cfg` CLI.
- Also had to tweak yaml2cfg functions to allow the display of mandatory parameters that do not follow the standard parameter scheme.

- [X] Nicely displays `postprocess` and `preprocess` parameters
- [X] Also display mandatory global parameters now (`rundir` and `molecules`)

Please try the following command now:
```bash
haddock3-cfg
```

### CODE REVIEW

File: src/haddock/gear/yaml2cfg.py [hunk 1]
```diff
@@ -45,7 +49,17 @@ def yaml2cfg_text(
         parameters will be considered.
 
     details : bool
-        Whether to add the 'long' description of each parameter.
+        Wheather to add the 'long' description of each parameter.
```
Comment (amjjbonvin) [inline src/haddock/gear/yaml2cfg.py:52]: Typo: Wheather -> Whether (i.e. was correct initially)

Comment (amjjbonvin) [reply src/haddock/gear/yaml2cfg.py:52]: And other instances to be corrected as well

Comment (VGPReys) [inline src/haddock/gear/yaml2cfg.py:52]:
```suggestion
        Whether to add the 'long' description of each parameter.
```

File: src/haddock/gear/yaml2cfg.py [hunk 2]
```diff
@@ -45,7 +49,17 @@ def yaml2cfg_text(
...
+    
+    mandatory_param : bool
+        Wheather this current parameters are mandatory ones. Special case where
```
Comment (VGPReys) [inline src/haddock/gear/yaml2cfg.py:55]:
```suggestion
        Whether this current parameters are mandatory ones. Special case where
```

File: src/haddock/gear/yaml2cfg.py [hunk 3]
```diff
@@ -89,6 +108,16 @@ def _yaml2cfg_text(
 
     details : bool
         Whether to add the 'long' description of each parameter.
+
+    mandatory_param : bool
+        Wheather this current parameters are mandatory ones. Special case where
```
Comment (VGPReys) [inline src/haddock/gear/yaml2cfg.py:113]:
```suggestion
        Whether this current parameters are mandatory ones. Special case where
```

File: src/haddock/gear/yaml2cfg.py [hunk 4]
```diff
@@ -97,69 +126,80 @@ def _yaml2cfg_text(
     exp_level_idx = exp_levels[explevel]
 
     # define set of undesired parameter keys
-    undesired = ("default", "explevel", "type")
+    undesired = ["default", "explevel", "type"]
```
Comment (VGPReys) [inline src/haddock/gear/yaml2cfg.py:129]: I was wondering if we want to still display the `group` in this `haddock3-cfg` output.
If not, we can add it here to remove it
```suggestion
    undesired = ["default", "explevel", "type", "group"]
```

**Review by amjjbonvin — DISMISSED** [review]

**Review by rversin — APPROVED** [review]

**Pull request merged**